### PR TITLE
[new release] yocaml (9 packages) (1.0.0)

### PR DESCRIPTION
**Pull request by dinosaure**
YOCaml It is a static site generator, mostly written in OCaml

- Project page: <a href="https://github.com/xhtmlboi/yocaml">https://github.com/xhtmlboi/yocaml</a>

##### CHANGES:

- First release of YOCaml

### CODE REVIEW

File: packages/yocaml_unix/yocaml_unix.1.0.0/opam [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+depends: [
+  "ocaml" { >= "4.11.1" }
+  "dune" { >= "2.8" }
+  "odoc" {with-doc}
+  "preface" { >= "1.0.0" }
+  "cryptokit" { >= "1.16.1" }
+  "logs" {>= "0.7.0" }
+  "conduit-lwt" { >= "4.0.0" }
+  "cohttp-lwt-unix" { >= "4.0.0" }
+  "yocaml" {= version}
```
Comment (avsm) [inline packages/yocaml_unix/yocaml_unix.1.0.0/opam:31]:
```suggestion
  "lwt" {>="5.3.0"}
  "yocaml" {= version}
```

for let* in Lwt.Syntax

**Comment by dinosaure** [comment]
I think the error is related to the `result` package (`Result.map_error` exists since OCaml 4.08), I don't remember how to fix that (probably a conflict with the `result` package?)

**Comment by xvw** [comment]
A quick fix on YOCaml part can be replacing `Result.map_error` by `Preface.Result.Bifunctor.map_snd`. It is a little bit heavy but it should work.

**Comment by hannesm** [comment]
What you can do to avoid the failure is a `conflict: [ "result" {< "1.5"}]` in the opam file (see #19880 for details if interested).

**Comment by dinosaure** [comment]
The error is unrelated to this release (it concerns `uunf.15.1.0` which does not compile with OCaml 4.14 (flambda+fp):
```
#=== ERROR while compiling uunf.15.1.0 ========================================#
# context              2.2.0~alpha3 | linux/x86_64 | ocaml-options-only-flambda-fp.1 | file:///home/opam/opam-repository
# path                 ~/.opam/4.14/.opam-switch/build/uunf.15.1.0
# command              ~/.opam/opam-init/hooks/sandbox.sh build ocaml pkg/pkg.ml build --dev-pkg false --with-uutf true --with-cmdliner false
# exit-code            1
# env-file             ~/.opam/log/uunf-7-c22e55.env
# output-file          ~/.opam/log/uunf-7-c22e55.out
### output ###
# ocamlfind ocamldep -modules src/uunf_fmt.ml > src/uunf_fmt.ml.depends
# ocamlfind ocamlc -c -g -bin-annot -safe-string -I src -I support -I test -o src/uunf_fmt.cmo src/uunf_fmt.ml
# ocamlfind ocamldep -modules src/uunf_tmap.ml > src/uunf_tmap.ml.depends
# ocamlfind ocamlc -c -g -bin-annot -safe-string -I src -I support -I test -o src/uunf_tmap.cmo src/uunf_tmap.ml
# ocamlfind ocamlopt -c -g -bin-annot -safe-string -I src -I support -I test -o src/uunf_fmt.cmx src/uunf_fmt.ml
# ocamlfind ocamldep -modules src/uunf_tmapbool.ml > src/uunf_tmapbool.ml.depends
# ocamlfind ocamlc -c -g -bin-annot -safe-string -I src -I support -I test -o src/uunf_tmapbool.cmo src/uunf_tmapbool.ml
# ocamlfind ocamldep -modules src/uunf_tmapbyte.ml > src/uunf_tmapbyte.ml.depends
# ocamlfind ocamlc -c -g -bin-annot -safe-string -I src -I support -I test -o src/uunf_tmapbyte.cmo src/uunf_tmapbyte.ml
# ocamlfind ocamldep -modules src/uunf_data.ml > src/uunf_data.ml.depends
# ocamlfind ocamlc -c -g -bin-annot -safe-string -I src -I support -I test -o src/uunf_data.cmo src/uunf_data.ml
# ocamlfind ocamlopt -c -g -bin-annot -safe-string -I src -I support -I test -o src/uunf_tmap.cmx src/uunf_tmap.ml
# ocamlfind ocamlopt -c -g -bin-annot -safe-string -I src -I support -I test -o src/uunf_tmapbool.cmx src/uunf_tmapbool.ml
# ocamlfind ocamlopt -c -g -bin-annot -safe-string -I src -I support -I test -o src/uunf_tmapbyte.cmx src/uunf_tmapbyte.ml
# ocamlfind ocamldep -modules src/uunf.ml > src/uunf.ml.depends
# ocamlfind ocamldep -modules src/uunf.mli > src/uunf.mli.depends
# ocamlfind ocamlc -c -g -bin-annot -safe-string -I src -I support -I test -o src/uunf.cmi src/uunf.mli
# ocamlfind ocamlopt -c -g -bin-annot -safe-string -I src -I support -I test -o src/uunf_data.cmx src/uunf_data.ml
# + ocamlfind ocamlopt -c -g -bin-annot -safe-string -I src -I support -I test -o src/uunf_data.cmx src/uunf_data.ml
# Fatal error: exception Stack overflow
# Command exited with code 2.
# pkg.ml: [ERROR] cmd ['ocamlbuild' '-use-ocamlfind' '-classic-display' '-j' '4' '-tag' 'debug'
#      '-build-dir' '_build' 'opam' 'pkg/META' 'CHANGES.md' 'LICENSE.md'
#      'README.md' 'src/uunf.a' 'src/uunf.cmxs' 'src/uunf.cmxa' 'src/uunf.cma'
#      'src/uunf_string.cmx' 'src/uunf_string.cmi' 'src/uunf_string.mli'
#      'src/uunf.cmx' 'src/uunf.cmi' 'src/uunf.mli' 'src/uunf_data.cmx'
#      'src/uunf_tmapbyte.cmx' 'src/uunf_tmapbool.cmx' 'src/uunf_tmap.cmx'
#      'src/uunf_fmt.cmx' 'doc/index.mld' 'test/examples.ml']: exited with 10
```

**Comment by mseri** [comment]
The uunf stack overflow is well known, there is nothing to do on your side
Thanks a lot for the release

**Pull request merged**